apis/core/lib/domain/dimension/DimensionMetadataCollection.ts

```diff
@@ -46,6 +47,16 @@ export default function Mixin<Base extends Constructor<Omit<DimensionMetadataCol
       }
     }
 
+    deleteDimension(dimension: DimensionMetadata.DimensionMetadata): void {
+      const found = this.find(dimension.pointer.out(schema.about).term!)
+      if (!found) {
+        throw new Error('Dimension not found')
+      }
+      found.pointer.deleteOut()
```

Comment:
Since we already have the `dimension`, I don't think it's useful to `.find` it again.

```suggestion
      dimension.pointer.deleteOut()
```

The other solution is to take a `NamedNode` in this method and to not `.find` the dimension it beforehand.